File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    string text = get_string("Text: ");
    int num_words, num_letters, num_sentences;
    num_words = num_letters = num_sentences = 0;
    
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            num_letters++;      //add one to num_letters if is a alphabet
        }
            
        if ((i == 0 && text[i] != ' ') || (i != len - 1 && text[i] == ' ' && text[i + 1] != ' '))
        {
            num_words++;        //add one to num_words if a sequence of letters is separated by spaces
        }
            
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            num_sentences++;        //add one to num_sentences if encounters one of '.', '?', or '!'
        }
    }
    float L = ((float) num_letters / (float) num_words) * 100;
    float S = ((float) num_sentences / (float) num_words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);              
    
    if (index < 1)
    {
        printf("Before Grade 1\n");        //returns "Before Grade 1" if index is less than 1
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");            //returns "Grade 16+" if index is greater or equal to 16
    }
    else
    {
        printf("Grade %i\n", index);      //returns returns "Grade %i" where %i is the place holder for index
    }
}

File: readability.py
import cs50

def main():
    text = cs50.get_string("Text: ")
    num_letters = 0
    num_words = 0
    num_sentences = 0
    
    """
    Coleman-Liau index
    """
    L = (float(count_letters(text, num_letters)) / float(count_words(text, num_words))) * 100.0;
    S = (float(count_sentences(text, num_sentences)) / float(count_words(text, num_words))) * 100.0;
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index >= 16:
        print("Geade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade " + str(index))


def count_letters(text, num_letters):
    """
    helper function for total number of letters
    """
    for char in text:
        if char.isalpha():
            num_letters += 1
    return num_letters
    
def count_words(text, num_words):
    """"
    helper function for total number of words
    """
    for i in range(len(text)):
        if text[i] == " ":
            num_words += 1
    return num_words + 1
    
def count_sentences(text, num_sentences):
    """
    helper function for total number of sentences
    """
    for i in range(len(text)):
        if text[i] == "." or text[i] == "?" or text[i] == "!":
            num_sentences += 1
    return num_sentences
    


if __name__ == "__main__":
    main()